pyiceberg/catalog/__init__.py

```diff
@@ -746,6 +747,24 @@ def _convert_schema_if_needed(schema: Union[Schema, "pa.Schema"]) -> Schema:
             pass
         raise ValueError(f"{type(schema)=}, but it must be pyiceberg.schema.Schema or pyarrow.Schema")
 
+    @staticmethod
+    def metadata_file_location(table_location: str, file_name: str, properties: Properties = EMPTY_DICT) -> str:
```

Comment:
nit: should this be a Table static method, instead of a Catalog one? I think Table makes sense to me esp since we're reading TableProperties